airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/data/iceberg/parquet/AirbyteTypeToIcebergSchema.kt

```diff
@@ -93,25 +93,27 @@ class AirbyteTypeToIcebergSchema {
 }
 
 fun ObjectType.toIcebergSchema(primaryKeys: List<List<String>>): Schema {
-    val mutableListOf = mutableListOf<NestedField>()
+    val fields = mutableListOf<NestedField>()
     val identifierFields = mutableSetOf<Int>()
     val identifierFieldNames = primaryKeys.flatten().toSet()
     val icebergTypeConverter = AirbyteTypeToIcebergSchema()
     this.properties.entries.forEach { (name, field) ->
         val id = generatedSchemaFieldId()
-        mutableListOf.add(
+        val isPrimaryKey = identifierFieldNames.contains(name)
+        val isOptional = !isPrimaryKey && field.nullable
```

Comment:
fyi we tried enforcing PK requiredness at one point and it caused a lot of problems (there's a surprising number of sources that emit null PKs). If iceberg supports null values in a partition key, we should do that instead

Reply:
It just wont allow you to mark a field as an identifier field if its nullable. Only non-nullable fields can be identifier fields

Reply:
ugh. I think we Officially Support composite PKs with nullable fields https://github.com/airbytehq/airbyte/issues/31926 (see also slack #p0-primay-keys-cannot-be-null)

so that's going to be awkward